modules/dxp/apps/multi-factor-authentication/multi-factor-authentication-fido2-web/src/main/resources/META-INF/resources/mfa_fido2_checker/setup.jsp

```diff
@@ -16,18 +16,81 @@
 
 <%@ include file="/init.jsp" %>
 
+<%
+List<MFAFIDO2CredentialEntry> mfaIDO2CredentialEntries = MFAFIDO2CredentialEntryLocalServiceUtil.getMFAFIDO2CredentialEntriesByUserId(themeDisplay.getUserId());
+%>
+
 <div id="<portlet:namespace />messageContainer"></div>
 
-<aui:button-row>
-	<clay:button
-		additionalProps='<%=
-			HashMapBuilder.put(
-				"pkccOptions", request.getAttribute(MFAFIDO2WebKeys.MFA_FIDO2_PKCC_OPTIONS)
-			).build()
-		%>'
-		label="button-register-a-fido2-authenticator"
-		propsTransformer="js/RegistrationTransformer"
+<c:choose>
+	<c:when test="<%= mfaIDO2CredentialEntries.size() < mfaFIDO2Configuration.allowedCredentialsPerUser() %>">
+		<aui:button-row>
+			<clay:button
+				additionalProps='<%=
+					HashMapBuilder.put(
+						"pkccOptions", request.getAttribute(MFAFIDO2WebKeys.MFA_FIDO2_PKCC_OPTIONS)
+					).build()
+				%>'
+				label="button-register-a-fido2-authenticator"
+				propsTransformer="js/RegistrationTransformer"
+			/>
+		</aui:button-row>
+	</c:when>
+	<c:otherwise>
+		<liferay-ui:message key="you-have-registered-the-maximum-number-of-allowed-authenticators" />
+
+		<aui:input name="mfaRemoveExistingSetup" type="hidden" value="<%= true %>" />
+
+		<button class="btn btn-danger" type="submit">
+			<liferay-ui:message key="button-remove-all-registered-fido2-authenticators" />
+		</button>
+	</c:otherwise>
+</c:choose>
+
+<aui:input name="responseJSON" showRequiredLabel="yes" type="hidden" />
```

Comment:
This line adds a form input. Please check:

    * The input is validated on the processing action

    * The backend is protected agains value forging

    

    Refer to documentation: